cmd/podman/volumes/import.go

```diff
@@ -0,0 +1,75 @@
+package volumes
+
+import (
+	"context"
+	"os"
+
+	"github.com/containers/podman/v3/cmd/podman/common"
+	"github.com/containers/podman/v3/cmd/podman/inspect"
+	"github.com/containers/podman/v3/cmd/podman/parse"
+	"github.com/containers/podman/v3/cmd/podman/registry"
+	"github.com/containers/podman/v3/pkg/domain/entities"
+	"github.com/containers/podman/v3/utils"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	importDescription = `Imports contents into a podman volume from specified tarball (.tar, .tar.gz, .tgz, .bzip, .tar.xz, .txz).`
+	importCommand     = &cobra.Command{
+		Annotations:       map[string]string{registry.EngineMode: registry.ABIMode},
+		Use:               "import VOLUME [SOURCE]",
+		Short:             "Import a tarball contents into a podman volume",
+		Long:              importDescription,
+		RunE:              importVol,
+		Args:              cobra.ExactArgs(2),
+		ValidArgsFunction: common.AutocompleteVolumes,
+		Example: `podman volume import my_vol /home/user/import.tar
+  cat ctr.tar | podman import volume my_vol -`,
+	}
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Command: importCommand,
+		Parent:  volumeCmd,
+	})
+}
+
+func importVol(cmd *cobra.Command, args []string) error {
+	var inspectOpts entities.InspectOptions
+	var tarFile *os.File
+	containerEngine := registry.ContainerEngine()
+	ctx := context.Background()
+	// create a slice of volumes since inspect expects slice as arg
+	volumes := []string{args[0]}
+	tarPath := args[1]
+
+	if tarPath != "-" {
+		err := parse.ValidateFileName(tarPath)
+		if err != nil {
+			return err
+		}
+
+		// open tar file
+		tarFile, err = os.Open(tarPath)
+		if err != nil {
+			return err
+		}
+	} else {
+		tarFile = os.Stdin
+	}
+
+	inspectOpts.Type = inspect.VolumeType
+	volumeData, _, err := containerEngine.VolumeInspect(ctx, volumes, inspectOpts)
+	if err != nil {
+		return err
+	}
+	if len(volumeData) < 1 {
+		return errors.New("no volume data found")
+	}
+	mountPoint := volumeData[0].VolumeConfigResponse.Mountpoint
```

Comment:
is there any remote possibility VolumeConfigResponse or Mountpoint could be nil?

Reply:
It cant be nil but possibility  can be that nothing is mounted there so in case of `export` we did a explicit check for **mountpoints** for `import` we can just move data to blank dir and container should see stuff.

Reply:
cc @mheon

Reply:
I honestly don't know. That's not the struct that Libpod outputs, so it looks like the intermediate layer is munging it somehow. I don't really know if that munging can ever produce a circumstance where it's nil?